### ISSUE TITLE
Error editing entry

### ISSUE THREAD

**Issue by GRbit** [replydev/cotp#225]
Thank you very much for this app, I find it very convenient and use it a lot.

Now to the problem, when I try to edit existing entry I receive following error:
```bash
[0] $ cotp edit -i 8 -l withdraw
Password: 
thread 'main' panicked at 'Mismatch between definition and access of `index`. Could not downcast to TypeId { t: 13834754221672687376 }, need to downcast to TypeId { t: 8562976148318468642 }
', $HOME/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-4.1.1/src/parser/error.rs:30:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted
```

Here is the same thing with RUST_BACKTRACE=full
```bash
[0] $ RUST_BACKTRACE=full cotp edit -i 8 -l withdraw
Password: 
thread 'main' panicked at 'Mismatch between definition and access of `index`. Could not downcast to TypeId { t: 13834754221672687376 }, need to downcast to TypeId { t: 8562976148318468642 }
', /home/grbit/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-4.1.1/src/parser/error.rs:30:9
stack backtrace:
   0:     0x5594bcd2e469 - <unknown>
   1:     0x5594bcccebbe - <unknown>
   2:     0x5594bcd06644 - <unknown>
   3:     0x5594bcd2f4ff - <unknown>
   4:     0x5594bcd2f0ff - <unknown>
   5:     0x5594bcd30168 - <unknown>
   6:     0x5594bcd2fc52 - <unknown>
   7:     0x5594bcd2fbbc - <unknown>
   8:     0x5594bcd2fb91 - <unknown>
   9:     0x5594bcc5c3f2 - <unknown>
  10:     0x5594bcc94bd4 - <unknown>
  11:     0x5594bcc65f55 - <unknown>
  12:     0x5594bcc9a8c7 - <unknown>
  13:     0x7fb8ad269d0a - __libc_start_main
  14:     0x5594bcc61bfa - <unknown>
  15:                0x0 - <unknown>
Aborted
```

Maybe I'm doing something in a wrong way, then please explain me how to edit entries in my cotp database.

**Comment by replydev** [comment]
Thank you for reporting! I can replicate and fix will come very soon

**Comment by replydev** [comment]
@GRbit #230 should fix the issue. If you can, please try this [debug artifact](https://github.com/replydev/cotp/suites/13071504460/artifacts/709011745).
Debug binaries use a local .cotp folder, so it will not override your real OTP database. Do not run this in your home otherwise the paths would coincide.

**Comment by GRbit** [comment]
@replydev 
Wow, that was fast!

I've tested a binary I've compiled from `fix/broken_edit_command` branch and it works as expected. I think the issue could be closed, v1.2.5 edits entry as expected.

**Comment by replydev** [comment]
The fix was very easy, my fault for not having integration test, i should make them and do some refactors after this release.

I think we can merge and release 1.2.5.

Thank you for reporting.
